src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1355,5 +1374,120 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
 
             Assert.StartsWith(expected, title);
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))] // No Notepad on Nano
+        [OuterLoop("Launches notepad")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void StartInfo_LoadUserProfile_And_UseCredentialsForNetworkingOnly_AreIncompatible()
+        {
+            ProcessStartInfo info = new ProcessStartInfo
+            {
+                UseShellExecute = true,
+                LoadUserProfile = true,
+                UseCredentialsForNetworkingOnly = true,
+                FileName = "notepad.exe",
+                Arguments = null,
+                WindowStyle = ProcessWindowStyle.Minimized
+            };
+
+            Assert.Throws<ArgumentException>("startInfo", () =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    Assert.False(process != null, $"Process started despite incompatible options {nameof(info.LoadUserProfile)} and {nameof(info.UseCredentialsForNetworkingOnly)} were enabled");
+                }
+            });
+        }
+
+        [ConditionalFact(nameof(IsAdmin_IsNotNano_RemoteExecutorIsSupported))] // Nano has no "netapi32.dll", Admin rights are required
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [OuterLoop("Requires admin privileges")]
+        public void TestUserNetworkCredentialsPropertiesOnWindows()
+        {
+            string testFilePath = GetTestFilePath();
+            Assert.False(string.IsNullOrWhiteSpace(testFilePath), $"Path to test file should not be empty: {testFilePath}");
+
+            string testFilePathRoot = Path.GetPathRoot(testFilePath);
+            string testFileUncPath = $"\\\\{Environment.MachineName}\\{testFilePath.Substring(0, testFilePathRoot.IndexOf(":"))}$\\{testFilePath.Replace(testFilePathRoot, "")}";
+            string testFileContent = Guid.NewGuid().ToString();
+            File.WriteAllText(testFilePath, testFileContent);
+
+            using WindowsTestAccount accountWithoutFilePermission = new WindowsTestAccount("testForDotnetRuntime");
```

Comment:
this ctor creates an account with given name: https://github.com/dotnet/runtime/blob/9342649430c19af4bc9fe0d870609778f244a721/src/libraries/Common/tests/TestUtilities/System/WindowsIdentityFixture.cs#L44

and right after it you are removing the account and creating it again using sys-calls directly.

I really like the idea of `WindowsTestAccount` here (I had no idea that it exists). I think that you should extend it with the possibility to specify user password in the ctor and stop using the sys-calls directly here.

If you don't mind and it's possible, please refactor the `TestUserCredentialsPropertiesOnWindows` test to use the same approach as well:

https://github.com/dotnet/runtime/blob/85fa287fa4dcae54519ea4c4420bba583198df98/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs#L468

Reply:
> this ctor creates an account with given name:
> 
> https://github.com/dotnet/runtime/blob/9342649430c19af4bc9fe0d870609778f244a721/src/libraries/Common/tests/TestUtilities/System/WindowsIdentityFixture.cs#L44
> 
> and right after it you are removing the account and creating it again using sys-calls directly.

The first test account is `"testForDotnetRuntime"`, and the second is `"testForDotnetNetwork`". It's much clearer in the refactored version with `WindowsIdentityFixture`.

> I really like the idea of `WindowsTestAccount` here (I had no idea that it exists). I think that you should extend it with the possibility to specify user password in the ctor and stop using the sys-calls directly here.

I decided to expose the generated password as a property. Since it's a test helper class, I don't think there's huge risk of leaking secrets here. This way, you don't have to implement password generation in each test that might need it.

I created a separate PR #77804 for the refactoring, and pulled it in here for immediate use.